kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -22,12 +22,25 @@
 
 import io.delta.kernel.internal.fs.Path;
 import io.delta.kernel.internal.util.FileNames;
+import io.delta.kernel.internal.util.Preconditions;
 
 /**
  * Metadata about Delta checkpoint.
  */
 public class CheckpointInstance
     implements Comparable<CheckpointInstance> {
+
+    public enum CheckpointFormat {
+        CLASSIC, MULTI_PART, V2;
```

Comment:
Can you add a comment here saying that the order is very important as the `compareTo` implementation relies on it.

Reply:
Added.